cmake/compilerOptions.cmake

```diff
@@ -111,3 +111,39 @@ if(NOT MSVC)
         endif()
     endif()
 endif()
+
+set(REQUIRE_EXPLICIT_LIBC_LINK 0)
+if(USE_GNU_LINK_FLAGS)
+    # OpenBSD apparently requires an explicit -lc if -Wl,--no-undefined
+    # is used, but that is NOT required on other platforms, and on even
+    # other platforms (e.g. MinGW) -lc might not even exist at all.
+    # Therefore detect if -lc works and if it is required, and omit it
+    # if it doesn't.
+    # Just running check_linker_flag() doesn't help here, because we
+    # actually need to reference a symbol from libc to cause an error.
+    # Use -shared in CMAKE_REQUIRED_LINK_OPTIONS because how symbols
+    # (esp. undefined ones) are resolved during linking can differ
+    # between shared libraries and executables.
+    set(TEST_LINK_NO_UNDEFINED_EXIMPLICIT_LIBC "
```

Comment:
Is HAVE_LINK_NO_UNDEFINED some lines avove still necessary?

Reply:
Potentially no, but then you'd have to restructure your CMake file quite a bit. (Especially since you have the version check for CMake in there, and otherwise always set `USE_GNU_LINK_FLAGS` to 1 if it's not an apple system.) I wanted to keep things as simple as possible to just fix the regression.

Reply:
Actually, with my new update (see below), I've now reworked the logic as you suggested.